std/codec.go

```diff
@@ -9,6 +12,12 @@ import (
 	vesting "github.com/cosmos/cosmos-sdk/x/auth/vesting/types"
 )
 
+// Temporary to solve https://github.com/cosmos/cosmos-sdk/pull/7442
+func init() {
+	tmjson.RegisterType((*tmcrypto.PublicKey)(nil), "tendermint.crypto.PublicKey")
+	tmjson.RegisterType((*tmcrypto.PublicKey_Ed25519)(nil), "tendermint.crypto.PublicKey_Ed25519")
+}
```

Comment:
@marbar3778 I might need your help here.

In the test I added in this PR, we do a `tmjson encode` of PublicKey_Ed25519, which gives the error described in #7401 . 

I added these 2 lines, and it works. But I'm pretty sure something needs to be done on the TM side:
1. Either move this `init()` inside `proto/tendermint/crypto`:
2. Or convert that PublicKey_Ed25519 into a PubKey somewhere, and tmjson encode the PubKey.

I can create a PR for 1 if that's desirable.